conn.go

```diff
@@ -1689,36 +1689,37 @@ func (c *Conn) querySystemLocal(ctx context.Context) *Iter {
 func (c *Conn) awaitSchemaAgreement(ctx context.Context) (err error) {
 	const localSchemas = "SELECT schema_version FROM system.local WHERE key='local'"
 
-	var versions map[string]struct{}
+	versions := make(map[string]struct{})
 	var schemaVersion string
 
 	endDeadline := time.Now().Add(c.session.cfg.MaxWaitSchemaAgreement)
 
 	for time.Now().Before(endDeadline) {
-		iter := c.querySystemPeers(ctx, c.host.version)
+		iter := &Iter{}
+		if !c.session.cfg.DisableHostLookup {
+			iter = c.querySystemPeers(ctx, c.host.version)
 
-		versions = make(map[string]struct{})
-
-		rows, err := iter.SliceMap()
-		if err != nil {
-			goto cont
-		}
-
-		for _, row := range rows {
-			host, err := c.session.hostInfoFromMap(row, &HostInfo{connectAddress: c.host.ConnectAddress(), port: c.session.cfg.Port})
+			rows, err := iter.SliceMap()
 			if err != nil {
 				goto cont
 			}
-			if !isValidPeer(host) || host.schemaVersion == "" {
-				c.logger.Printf("invalid peer or peer with empty schema_version: peer=%q", host)
-				continue
-			}
 
-			versions[host.schemaVersion] = struct{}{}
-		}
+			for _, row := range rows {
+				host, err := c.session.hostInfoFromMap(row, &HostInfo{connectAddress: c.host.ConnectAddress(), port: c.session.cfg.Port})
+				if err != nil {
+					goto cont
+				}
+				if !isValidPeer(host) || host.schemaVersion == "" {
+					c.logger.Printf("invalid peer or peer with empty schema_version: peer=%q", host)
+					continue
+				}
 
-		if err = iter.Close(); err != nil {
-			goto cont
+				versions[host.schemaVersion] = struct{}{}
+			}
+
+			if err = iter.Close(); err != nil {
+				goto cont
+			}
```

Comment:
I think it's fine if we query system.peers for the purpose of retrieving the schema version of each host as long as we don't update gocql's ring/hosts. If a user wants to use a static set of contact points and avoid using hosts from system tables there's no reason why they would have to give up the functionality of awaiting for schema agreement.